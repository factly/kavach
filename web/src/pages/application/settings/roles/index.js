import React from 'react';
import { Skeleton, Space, Button } from 'antd';
import { useDispatch, useSelector } from 'react-redux';
import { Link, useParams } from 'react-router-dom';
import ApplicationRoleList from './components/RoleList';
import { PlusOutlined } from '@ant-design/icons';
import { getApplication } from '../../../../actions/application';
import { getApplicationRoles } from '../../../../actions/roles';

function ApplicationRoles() {
  const { id } = useParams();
  const dispatch = useDispatch();
  React.useEffect(() => {
    dispatch(getApplication(id));
    dispatch(getApplicationRoles(id));
  }, [id, dispatch]);

  const { application, loadingApp, role, loading } = useSelector((state) => {
    return {
      application: state.applications.details[id],
      loadingApp: state.applications.loading,
      role: state.profile.roles[state.organisations.selected],
      loading: state.profile.loading,
    };
  });

  return (
    <Space direction="vertical">
      {loading && loadingApp ? (
        <Skeleton />
      ) : (
        <>
          <div className="application-descriptions-header">
            <div className="application-descriptions-title">
              <h2 className="application-title-main">Roles</h2>
            </div>
            {role === 'owner' ? (
              <div>
                <Link
                  key="1"
                  to={{
                    pathname: `/applications/${id}/settings/roles/create`,
                  }}
                >
                  <Button icon={<PlusOutlined />} type="primary">
                    Create New Role
                  </Button>
                </Link>
              </div>
            ) : null}
          </div>
          <ApplicationRoleList appID={id} role={role} />
        </>
      )}
    </Space>
  );
}

export default ApplicationRoles;
