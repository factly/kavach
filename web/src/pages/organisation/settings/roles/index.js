import React from 'react';
import { Skeleton, Space, Button } from 'antd';
import { useDispatch, useSelector } from 'react-redux';
import { Link, useParams } from 'react-router-dom';
import { getOrganisation } from '../../../../actions/organisations';
import OrganisationRoleList from './components/RoleList';

function OrganisationRoles() {
  const { orgID } = useParams();
  const dispatch = useDispatch();
  React.useEffect(() => {
    dispatch(getOrganisation(orgID));
  }, [orgID, dispatch]);

  const { loadingOrg, role, loading } = useSelector((state) => {
    return {
      loadingOrg: state.organisations.loading,
      role: state.profile.roles[state.organisations.selected],
      loading: state.profile.loading,
    };
  });

  return (
    <Space direction="vertical">
      <Link key="1" to={`/organisation`}>
        <Button type="primary"> Back to Settings </Button>
      </Link>
      {loading || loadingOrg ? (
        <Skeleton />
      ) : (
        <Space direction="vertical">
          {role === 'owner' ? (
            <div style={{ display: 'flex', justifyContent: 'end' }}>
              <Link
                key="2"
                to={{
                  pathname: `/organisation/${orgID}/settings/roles/create`,
                }}
              >
                <Button type="primary"> Create New Role </Button>
              </Link>
            </div>
          ) : null}
          <OrganisationRoleList orgID={orgID} role={role} />
        </Space>
      )}
    </Space>
  );
}

export default OrganisationRoles;
