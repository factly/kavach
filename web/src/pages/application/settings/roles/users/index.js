import React from 'react';
import { Space, Form, Button, Select, Skeleton } from 'antd';
import { useParams } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { getApplicationUsers } from '../../../../../actions/applicationUsers';
import { addApplicationRoleUserByID, getApplicationRoleUsers } from '../../../../../actions/roles';
import UserList from './components/ApplicationRoleUserList';
import { Link } from 'react-router-dom';

export default function ApplicationRoleUsers() {
  const [form] = Form.useForm();
  const dispatch = useDispatch();
  const { appID, roleID } = useParams();
  const { roleUsers, remainingAppUsers, loading, userRole, loadingUserRole } = useSelector(
    (state) => {
      var remUserIDs = [];
      var roleUserIDs = [];
      roleUserIDs = state.roles.application?.[appID]?.[roleID].users || [];
      const appUserIDs = state.applications.details[appID]?.users || [];
      remUserIDs = appUserIDs.filter((uID) => roleUserIDs.every((rUID) => !(rUID === uID)));
      return {
        roleUsers: roleUserIDs.map((id) => state.users.details[id]),
        remainingAppUsers: remUserIDs.map((id) => state.users.details[id]),
        loading: state.roles.loading,
        userRole: state.profile.roles[state.organisations.selected],
        loadingUserRole: state.profile.loading,
      };
    },
  );

  const fetchApplicationUsers = () => {
    dispatch(getApplicationUsers(appID));
  };

  const fetchApplicationRoleUsers = () => {
    dispatch(getApplicationRoleUsers(appID, roleID));
  };

  const onReset = () => {
    form.resetFields();
  };

  const onSubmit = (values) => {
    dispatch(addApplicationRoleUserByID(appID, roleID, values.user_id)).then(() =>
      dispatch(getApplicationRoleUsers(appID, roleID)),
    );
    onReset();
  };

  React.useEffect(() => {
    fetchApplicationUsers();
    fetchApplicationRoleUsers();
    // eslint-disable-next-line
  }, []);

  return (
    <div
      style={{
        display: 'flex',
        flexDirection: 'column',
        gap: '20px',
      }}
    >
      <Link key="1" to={`/applications/${appID}/settings/roles`}>
        <Button type="primary">Back to Roles</Button>
      </Link>
      {loading && loadingUserRole ? (
        <Skeleton />
      ) : (
        <Space direction="vertical">
          {userRole === 'owner' ? (
            <Form
              form={form}
              name="add-application-role-user"
              layout="inline"
              onFinish={(values) => {
                onSubmit(values);
              }}
            >
              <Form.Item name="user_id" label="Users">
                <Select bordered listHeight={128} style={{ width: 200 }} placeholder="select user">
                  {remainingAppUsers.map((user, index) => (
                    <Select.Option value={user.id} key={index}>
                      {user.email}
                    </Select.Option>
                  ))}
                </Select>
              </Form.Item>
              <Form.Item>
                <Button type="primary" htmlType="submit" form="add-application-role-user">
                  Add Users
                </Button>
              </Form.Item>
            </Form>
          ) : null}
          <UserList appID={appID} roleID={roleID} users={roleUsers} />
        </Space>
      )}
    </div>
  );
}
