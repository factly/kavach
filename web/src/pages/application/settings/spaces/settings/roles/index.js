import React from 'react';
import { Skeleton, Space, Button } from 'antd';
import { useDispatch, useSelector } from 'react-redux';
import { Link, useParams } from 'react-router-dom';
import SpaceRoleList from './components/RoleList';
import { getSpaceByID } from '../../../../../../actions/space';
import { getSpaceRoles } from '../../../../../../actions/roles';

export default function SpaceRoles() {
  const { appID, spaceID } = useParams();
  const dispatch = useDispatch();

  React.useEffect(() => {
    dispatch(getSpaceRoles(appID, spaceID));
  }, [dispatch, appID, spaceID]);

  const { space, loadingSpace, role, loading } = useSelector((state) => {
    return {
      space: state.spaces.details[spaceID],
      loadingSpace: state.spaces.loading,
      role: state.profile.roles[state.organisations.selected],
      loading: state.profile.loading,
    };
  });

  return (
    <div
      style={{
        display:'flex',
        flexDirection:'column',
        gap:'20px'
      }}
    >
      <Link key="1" to={`/applications/${appID}/settings/spaces/${spaceID}/settings`}>
        <Button type="primary"> Back to Settings </Button>
      </Link>
      <h2>Roles in {space?.name}</h2>
      {loading && loadingSpace ? (
        <Skeleton />
      ) : (
        <Space direction="vertical">
          {role === 'owner' ? (
            <div style={{ display: 'flex', justifyContent: 'end' }}>
              <Link
                key="1"
                to={{
                  pathname: `/applications/${appID}/settings/spaces/${spaceID}/settings/roles/create`,
                }}
              >
                <Button type="primary"> Create New Role </Button>
              </Link>
            </div>
          ) : null}
          <SpaceRoleList appID={appID} role={role} spaceID={spaceID} />
        </Space>
      )}
    </div>
  );
}
