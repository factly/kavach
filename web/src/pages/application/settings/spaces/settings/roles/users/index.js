import React from 'react';
import { Space, Form, Button, Select, Skeleton } from 'antd';
import { Link, useParams } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { getSpaceUsers } from '../../../../../../../actions/spaceUser';
import { addSpaceRoleUserByID, getSpaceRoleUsers } from '../../../../../../../actions/roles';
import UserList from './components/SpaceRoleUserList';

export default function SpaceRoleUsers() {
  const [form] = Form.useForm();
  const dispatch = useDispatch();
  const { appID, spaceID, roleID } = useParams();
  const { roleUsers, remainingSpaceUsers, loading, userRole, loadingUserRole } = useSelector(
    (state) => {
      var remUserIDs = [];
      var roleUserIDs = [];
      roleUserIDs = state.roles.space[spaceID][roleID]?.users || [];
      const spaceUserIDs = state.spaces.details[spaceID]?.users || [];
      if (spaceUserIDs?.length) {
        remUserIDs = spaceUserIDs.filter((uID) => roleUserIDs.every((rUID) => !(rUID === uID)));
      }
      return {
        roleUsers: roleUserIDs.map((id) => state.users.details[id]),
        remainingSpaceUsers: remUserIDs.map((id) => state.users.details[id]),
        loading: state.roles.loading,
        userRole: state.profile.roles[state.organisations.selected],
        loadingUserRole: state.profile.loading,
      };
    },
  );

  const fetchSpaceUsers = () => {
    dispatch(getSpaceUsers(appID, spaceID));
  };

  const fetchSpaceRoleUsers = () => {
    dispatch(getSpaceRoleUsers(appID, spaceID, roleID));
  };

  const onReset = () => {
    form.resetFields();
  };

  const onSubmit = (values) => {
    dispatch(addSpaceRoleUserByID(appID, spaceID, roleID, values.user_id)).then(() =>
      dispatch(getSpaceRoleUsers(appID, spaceID, roleID)),
    );
    onReset();
  };

  React.useEffect(() => {
    fetchSpaceUsers();
    fetchSpaceRoleUsers();
    // eslint-disable-next-line
  }, []);

  return (
    <div
      style={{
        display: 'flex',
        flexDirection: 'column',
        gap: '20px',
      }}
    >
      <Link key="1" to={`/applications/${appID}/settings/spaces/${spaceID}/settings/roles`}>
        <Button type="primary">Back to Roles</Button>
      </Link>
      {loading && loadingUserRole ? (
        <Skeleton />
      ) : (
        <Space direction="vertical">
          {userRole === 'owner' ? (
            <Form
              form={form}
              name="add-space-role-user"
              layout="inline"
              onFinish={(values) => {
                onSubmit(values);
              }}
            >
              <Form.Item name="user_id" label="Users">
                <Select bordered listHeight={128} style={{ width: 200 }} placeholder="select user">
                  {remainingSpaceUsers.map((user, index) => (
                    <Select.Option value={user.id} key={index}>
                      {user.email}
                    </Select.Option>
                  ))}
                </Select>
              </Form.Item>
              <Form.Item>
                <Button type="primary" htmlType="submit" form="add-space-role-user">
                  Add Users
                </Button>
              </Form.Item>
            </Form>
          ) : null}
          <UserList appID={appID} spaceID={spaceID} roleID={roleID} users={roleUsers} />
        </Space>
      )}
    </div>
  );
}
