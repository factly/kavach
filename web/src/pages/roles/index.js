import React from 'react';
import { Collapse, Form, Tabs } from 'antd';
import FormComponent from './components/RolesForm';
import { useDispatch } from 'react-redux';
import { getOrganisationRoles, getApplicationRoles, getSpaceRoles,  createOrganisationRole, createApplicationRole, createSpaceRole } from '../../actions/roles';

function Roles() {
  const [form] = Form.useForm();
  const dispatch = useDispatch();
  const onSubmit = (value, type) => {
    const reqBody = {
      name: value.name,
      description: value.description,
    }
    switch (type) {
      case 'organisation':
        dispatch(createOrganisationRole(reqBody)).then(() => dispatch(getOrganisationRoles()));
        break;
      case 'application':
        dispatch(createApplicationRole(reqBody)).then(() => getApplicationRoles(value.application));
        break;
      case 'space':
        dispatch(createSpaceRole(reqBody)).then(() =>
          dispatch(getSpaceRoles(value.application, value.space)),
        );
        break;
      default:
        return;
    }
    form.resetFields();
  };

  return (
    <Collapse>
      <Collapse.Panel header=" Create Roles">
        <Tabs centered={true}>
          <Tabs.TabPane tab="Organisation roles" key="1">
            <FormComponent form={form} onSubmit={onSubmit} type="organisation" />
          </Tabs.TabPane>
          <Tabs.TabPane tab="Application roles" key="2">
            <FormComponent form={form} onSubmit={onSubmit} type="application" />
          </Tabs.TabPane>
          <Tabs.TabPane tab="Space roles" key="3">
            <FormComponent form={form} onSubmit={onSubmit} type="space" />
          </Tabs.TabPane>
        </Tabs>
      </Collapse.Panel>
      <Collapse.Panel header=" View Roles">
        <Tabs centered={true}>
          <Tabs.TabPane tab="Organisation roles" key="1"></Tabs.TabPane>
          <Tabs.TabPane tab="Application roles" key="2"></Tabs.TabPane>
          <Tabs.TabPane tab="Space roles" key="3"></Tabs.TabPane>
        </Tabs>
      </Collapse.Panel>
    </Collapse>
  );
}

export default Roles;
