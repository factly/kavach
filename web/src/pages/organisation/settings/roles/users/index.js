import React from 'react';
import { Space, Form, Button, Select, Skeleton } from 'antd';
import { Link, useParams } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { getUsers } from '../../../../../actions/users';
import {
  addOrganisationRoleUserByID,
  getOrganisationRoleUsers,
} from '../../../../../actions/roles';
import UserList from './components/OrganisationRoleUserList';

export default function OrganisationRoleUsers() {
  const [form] = Form.useForm();
  const dispatch = useDispatch();
  const { orgID, roleID } = useParams();

  const { roleUsers, remainingOrgUsers, loading, userRole, loadingUserRole } = useSelector(
    (state) => {
      var remUserIDs = [];
      var roleUserIDs = [];
      roleUserIDs = state.roles.organisation[orgID][roleID].users || [];
      const orgUserIDs = state.organisations.details[orgID].users || [];
      remUserIDs = orgUserIDs.filter((uID) => roleUserIDs.every((rUID) => !(rUID === uID)));
      return {
        roleUsers: roleUserIDs.map((id) => state.users.details[id]),
        remainingOrgUsers: remUserIDs.map((id) => state.users.details[id]),
        loading: state.roles.loading,
        userRole: state.profile.roles[state.organisations.selected],
        loadingUserRole: state.profile.loading,
      };
    },
  );

  const fetchOrganisationUsers = () => {
    dispatch(getUsers(roleID));
  };

  const fetchOrganisationRoleUsers = () => {
    dispatch(getOrganisationRoleUsers(roleID));
  };

  const onReset = () => {
    form.resetFields();
  };

  const onSubmit = (values) => {
    dispatch(addOrganisationRoleUserByID(roleID, values.user_id)).then(() =>
      dispatch(getOrganisationRoleUsers(roleID)),
    );
    onReset();
  };

  React.useEffect(() => {
    fetchOrganisationRoleUsers();
    fetchOrganisationUsers();
    //eslint-disable-next-line
  }, [orgID, roleID, dispatch]);
  return (
    <div
      style={{
        display: 'flex',
        flexDirection: 'column',
        gap: '20px',
      }}
    >
      <Link key="1" to={`/organisation/${orgID}/settings/roles`}>
        <Button type="primary">Back to Roles</Button>
      </Link>
      {loading || loadingUserRole ? (
        <Skeleton />
      ) : userRole !== 'owner' ? (
        <UserList roleID={roleID} users={roleUsers} />
      ) : (
        <Space direction="vertical">
          <Form
            form={form}
            name="add-organisation-role-user"
            layout="inline"
            onFinish={(values) => {
              onSubmit(values);
            }}
          >
            <Form.Item name="user_id" label="Users">
              <Select bordered listHeight={128} style={{ width: 200 }} placeholder="select user">
                {remainingOrgUsers.map((user, index) => (
                  <Select.Option value={user?.id} key={index}>
                    {user?.email}
                  </Select.Option>
                ))}
              </Select>
            </Form.Item>
            <Form.Item>
              <Button type="primary" htmlType="submit" form="add-organisation-role-user">
                Add Users
              </Button>
            </Form.Item>
          </Form>
          <UserList roleID={roleID} users={roleUsers} />
        </Space>
      )}
    </div>
  );
}
