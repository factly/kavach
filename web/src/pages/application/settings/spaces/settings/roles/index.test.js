import React from 'react';
import { useHistory } from 'react-router-dom';
import { useDispatch, Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import { BrowserRouter as Router } from 'react-router-dom';
import thunk from 'redux-thunk';
import { mount } from 'enzyme';
import SpaceRoles from '.';

import '../../../../../../matchMedia.mock';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

jest.mock('react-redux', () => ({
	...jest.requireActual('react-redux'),
	useDispatch: jest.fn(),
}));

jest.mock('react-router-dom', () => ({
	...jest.requireActual('react-router-dom'),
	useHistory: jest.fn(),
	useParams: jest.fn(() => {
		return {
			appID: 1,
			spaceID: 1,
		};
	}),
}));

import { getSpaceRoles } from '../../../../../../actions/roles';

jest.mock('../../../../../../actions/roles', () => ({
	getSpaceRoles: jest.fn(),
}));

jest.mock('./components/RoleList', () => {
	return jest.fn((props) => <div {...props}>SpaceRoleList</div>);
})

// space: state.spaces.details[spaceID],
// loadingSpace: state.spaces.loading,
// role: state.profile.roles[state.organisations.selected],
// loading: state.profile.loading,

let state = {
	spaces: {
		details: {
			1: {
				id: 1, name: 'Test Space', description: 'Test Space Description',
			},
		},
		loading: false,
	},
	profile: {
		roles: {
			1: 'owner',
		},
		loading: false,
	},
	organisations: {
		selected: 1,
	},
};
let store;
const mockedDispatch = jest.fn();
mockedDispatch.mockReturnValue(Promise.resolve());
useDispatch.mockReturnValue(mockedDispatch);
store = mockStore({});
const push = jest.fn();
useHistory.mockReturnValue({ push });

describe('Space Roles Component', () => {
	it('should match snapshot when loading', () => {
		state.spaces.loading = true;
		store = mockStore(state);
		const tree = mount(
			<Provider store={store}>
				<Router>
					<SpaceRoles />
				</Router>
			</Provider>,
		);
		expect(tree).toMatchSnapshot();
		expect(tree.find('Skeleton').length).toBe(1);
	})
	it('should match snapshot when loading roles', () => {
		state.spaces.loading = false;
		state.profile.loading = true;
		store = mockStore(state);
		const tree = mount(
			<Provider store={store}>
				<Router>
					<SpaceRoles />
				</Router>
			</Provider>,
		);
		expect(tree).toMatchSnapshot();
		expect(tree.find('Skeleton').length).toBe(1);
	})
	it('should match snapshot when no role is not owner', () => {
		state.spaces.loading = false;
		state.profile.loading = false;
		state.profile.roles[1] = 'admin';
		store = mockStore(state);
		const tree = mount(
			<Provider store={store}>
				<Router>
					<SpaceRoles />
				</Router>
			</Provider>,
		);
		expect(tree).toMatchSnapshot();
		expect(tree.find('Link').length).toBe(1);
	})

	it('should match snapshot when role is owner', () => {
		state.profile.roles[1] = 'owner';
		store = mockStore(state);
		const tree = mount(
			<Provider store={store}>
				<Router>
					<SpaceRoles />
				</Router>
			</Provider>,
		);
		expect(tree).toMatchSnapshot();
		expect(tree.find('Link').at(1).props().to).toEqual({
			pathname: '/applications/1/settings/spaces/1/settings/roles/create',
		})
	})
})
