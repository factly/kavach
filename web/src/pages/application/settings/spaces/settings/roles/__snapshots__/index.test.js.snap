// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Space Roles Component should match snapshot when loading 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <SpaceRoles>
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "column",
              "gap": "20px",
            }
          }
        >
          <Link
            key="1"
            to="/applications/1/settings/spaces/1/settings"
          >
            <LinkAnchor
              href="/applications/1/settings/spaces/1/settings"
              navigate={[Function]}
            >
              <a
                href="/applications/1/settings/spaces/1/settings"
                onClick={[Function]}
              >
                <Button
                  type="primary"
                >
                  <Wave
                    disabled={false}
                  >
                    <button
                      className="ant-btn ant-btn-primary"
                      disabled={false}
                      onClick={[Function]}
                      type="button"
                    >
                      <LoadingIcon
                        existIcon={false}
                        loading={false}
                        prefixCls="ant-btn"
                      >
                        <CSSMotion
                          motionName="ant-btn-loading-icon-motion"
                          onAppearActive={[Function]}
                          onAppearStart={[Function]}
                          onEnterActive={[Function]}
                          onEnterStart={[Function]}
                          onLeaveActive={[Function]}
                          onLeaveStart={[Function]}
                          removeOnLeave={true}
                          visible={false}
                        >
                          <DomWrapper />
                        </CSSMotion>
                      </LoadingIcon>
                      <span
                        key=".0"
                      >
                         Back to Settings 
                      </span>
                    </button>
                  </Wave>
                </Button>
              </a>
            </LinkAnchor>
          </Link>
          <h2>
            Roles in 
            Test Space
          </h2>
          <Skeleton>
            <div
              className="ant-skeleton"
            >
              <div
                className="ant-skeleton-content"
              >
                <Title
                  prefixCls="ant-skeleton-title"
                  width="38%"
                >
                  <h3
                    className="ant-skeleton-title"
                    style={
                      Object {
                        "width": "38%",
                      }
                    }
                  />
                </Title>
                <Paragraph
                  prefixCls="ant-skeleton-paragraph"
                  rows={3}
                  width="61%"
                >
                  <ul
                    className="ant-skeleton-paragraph"
                  >
                    <li
                      key="0"
                      style={
                        Object {
                          "width": undefined,
                        }
                      }
                    />
                    <li
                      key="1"
                      style={
                        Object {
                          "width": undefined,
                        }
                      }
                    />
                    <li
                      key="2"
                      style={
                        Object {
                          "width": "61%",
                        }
                      }
                    />
                  </ul>
                </Paragraph>
              </div>
            </div>
          </Skeleton>
        </div>
      </SpaceRoles>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`Space Roles Component should match snapshot when loading roles 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <SpaceRoles>
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "column",
              "gap": "20px",
            }
          }
        >
          <Link
            key="1"
            to="/applications/1/settings/spaces/1/settings"
          >
            <LinkAnchor
              href="/applications/1/settings/spaces/1/settings"
              navigate={[Function]}
            >
              <a
                href="/applications/1/settings/spaces/1/settings"
                onClick={[Function]}
              >
                <Button
                  type="primary"
                >
                  <Wave
                    disabled={false}
                  >
                    <button
                      className="ant-btn ant-btn-primary"
                      disabled={false}
                      onClick={[Function]}
                      type="button"
                    >
                      <LoadingIcon
                        existIcon={false}
                        loading={false}
                        prefixCls="ant-btn"
                      >
                        <CSSMotion
                          motionName="ant-btn-loading-icon-motion"
                          onAppearActive={[Function]}
                          onAppearStart={[Function]}
                          onEnterActive={[Function]}
                          onEnterStart={[Function]}
                          onLeaveActive={[Function]}
                          onLeaveStart={[Function]}
                          removeOnLeave={true}
                          visible={false}
                        >
                          <DomWrapper />
                        </CSSMotion>
                      </LoadingIcon>
                      <span
                        key=".0"
                      >
                         Back to Settings 
                      </span>
                    </button>
                  </Wave>
                </Button>
              </a>
            </LinkAnchor>
          </Link>
          <h2>
            Roles in 
            Test Space
          </h2>
          <Skeleton>
            <div
              className="ant-skeleton"
            >
              <div
                className="ant-skeleton-content"
              >
                <Title
                  prefixCls="ant-skeleton-title"
                  width="38%"
                >
                  <h3
                    className="ant-skeleton-title"
                    style={
                      Object {
                        "width": "38%",
                      }
                    }
                  />
                </Title>
                <Paragraph
                  prefixCls="ant-skeleton-paragraph"
                  rows={3}
                  width="61%"
                >
                  <ul
                    className="ant-skeleton-paragraph"
                  >
                    <li
                      key="0"
                      style={
                        Object {
                          "width": undefined,
                        }
                      }
                    />
                    <li
                      key="1"
                      style={
                        Object {
                          "width": undefined,
                        }
                      }
                    />
                    <li
                      key="2"
                      style={
                        Object {
                          "width": "61%",
                        }
                      }
                    />
                  </ul>
                </Paragraph>
              </div>
            </div>
          </Skeleton>
        </div>
      </SpaceRoles>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`Space Roles Component should match snapshot when no role is not owner 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <SpaceRoles>
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "column",
              "gap": "20px",
            }
          }
        >
          <Link
            key="1"
            to="/applications/1/settings/spaces/1/settings"
          >
            <LinkAnchor
              href="/applications/1/settings/spaces/1/settings"
              navigate={[Function]}
            >
              <a
                href="/applications/1/settings/spaces/1/settings"
                onClick={[Function]}
              >
                <Button
                  type="primary"
                >
                  <Wave
                    disabled={false}
                  >
                    <button
                      className="ant-btn ant-btn-primary"
                      disabled={false}
                      onClick={[Function]}
                      type="button"
                    >
                      <LoadingIcon
                        existIcon={false}
                        loading={false}
                        prefixCls="ant-btn"
                      >
                        <CSSMotion
                          motionName="ant-btn-loading-icon-motion"
                          onAppearActive={[Function]}
                          onAppearStart={[Function]}
                          onEnterActive={[Function]}
                          onEnterStart={[Function]}
                          onLeaveActive={[Function]}
                          onLeaveStart={[Function]}
                          removeOnLeave={true}
                          visible={false}
                        >
                          <DomWrapper />
                        </CSSMotion>
                      </LoadingIcon>
                      <span
                        key=".0"
                      >
                         Back to Settings 
                      </span>
                    </button>
                  </Wave>
                </Button>
              </a>
            </LinkAnchor>
          </Link>
          <h2>
            Roles in 
            Test Space
          </h2>
          <Space
            direction="vertical"
          >
            <div
              className="ant-space ant-space-vertical"
              style={Object {}}
            >
              <Item
                className="ant-space-item"
                direction="vertical"
                index={0}
                key="ant-space-item-0"
                marginDirection="marginRight"
                wrap={false}
              />
              <Item
                className="ant-space-item"
                direction="vertical"
                index={1}
                key="ant-space-item-1"
                marginDirection="marginRight"
                wrap={false}
              >
                <div
                  className="ant-space-item"
                  style={Object {}}
                >
                  <mockConstructor
                    appID={1}
                    role="admin"
                    spaceID={1}
                  >
                    <div
                      appID={1}
                      role="admin"
                      spaceID={1}
                    >
                      SpaceRoleList
                    </div>
                  </mockConstructor>
                </div>
              </Item>
            </div>
          </Space>
        </div>
      </SpaceRoles>
    </Router>
  </BrowserRouter>
</Provider>
`;

exports[`Space Roles Component should match snapshot when role is owner 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <SpaceRoles>
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "column",
              "gap": "20px",
            }
          }
        >
          <Link
            key="1"
            to="/applications/1/settings/spaces/1/settings"
          >
            <LinkAnchor
              href="/applications/1/settings/spaces/1/settings"
              navigate={[Function]}
            >
              <a
                href="/applications/1/settings/spaces/1/settings"
                onClick={[Function]}
              >
                <Button
                  type="primary"
                >
                  <Wave
                    disabled={false}
                  >
                    <button
                      className="ant-btn ant-btn-primary"
                      disabled={false}
                      onClick={[Function]}
                      type="button"
                    >
                      <LoadingIcon
                        existIcon={false}
                        loading={false}
                        prefixCls="ant-btn"
                      >
                        <CSSMotion
                          motionName="ant-btn-loading-icon-motion"
                          onAppearActive={[Function]}
                          onAppearStart={[Function]}
                          onEnterActive={[Function]}
                          onEnterStart={[Function]}
                          onLeaveActive={[Function]}
                          onLeaveStart={[Function]}
                          removeOnLeave={true}
                          visible={false}
                        >
                          <DomWrapper />
                        </CSSMotion>
                      </LoadingIcon>
                      <span
                        key=".0"
                      >
                         Back to Settings 
                      </span>
                    </button>
                  </Wave>
                </Button>
              </a>
            </LinkAnchor>
          </Link>
          <h2>
            Roles in 
            Test Space
          </h2>
          <Space
            direction="vertical"
          >
            <div
              className="ant-space ant-space-vertical"
              style={Object {}}
            >
              <Item
                className="ant-space-item"
                direction="vertical"
                index={0}
                key="ant-space-item-0"
                marginDirection="marginRight"
                wrap={false}
              >
                <div
                  className="ant-space-item"
                  style={
                    Object {
                      "marginBottom": 8,
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "display": "flex",
                        "justifyContent": "end",
                      }
                    }
                  >
                    <Link
                      key="1"
                      to={
                        Object {
                          "pathname": "/applications/1/settings/spaces/1/settings/roles/create",
                        }
                      }
                    >
                      <LinkAnchor
                        href="/applications/1/settings/spaces/1/settings/roles/create"
                        navigate={[Function]}
                      >
                        <a
                          href="/applications/1/settings/spaces/1/settings/roles/create"
                          onClick={[Function]}
                        >
                          <Button
                            type="primary"
                          >
                            <Wave
                              disabled={false}
                            >
                              <button
                                className="ant-btn ant-btn-primary"
                                disabled={false}
                                onClick={[Function]}
                                type="button"
                              >
                                <LoadingIcon
                                  existIcon={false}
                                  loading={false}
                                  prefixCls="ant-btn"
                                >
                                  <CSSMotion
                                    motionName="ant-btn-loading-icon-motion"
                                    onAppearActive={[Function]}
                                    onAppearStart={[Function]}
                                    onEnterActive={[Function]}
                                    onEnterStart={[Function]}
                                    onLeaveActive={[Function]}
                                    onLeaveStart={[Function]}
                                    removeOnLeave={true}
                                    visible={false}
                                  >
                                    <DomWrapper />
                                  </CSSMotion>
                                </LoadingIcon>
                                <span
                                  key=".0"
                                >
                                   Create New Role 
                                </span>
                              </button>
                            </Wave>
                          </Button>
                        </a>
                      </LinkAnchor>
                    </Link>
                  </div>
                </div>
              </Item>
              <Item
                className="ant-space-item"
                direction="vertical"
                index={1}
                key="ant-space-item-1"
                marginDirection="marginRight"
                wrap={false}
              >
                <div
                  className="ant-space-item"
                  style={Object {}}
                >
                  <mockConstructor
                    appID={1}
                    role="owner"
                    spaceID={1}
                  >
                    <div
                      appID={1}
                      role="owner"
                      spaceID={1}
                    >
                      SpaceRoleList
                    </div>
                  </mockConstructor>
                </div>
              </Item>
            </div>
          </Space>
        </div>
      </SpaceRoles>
    </Router>
  </BrowserRouter>
</Provider>
`;
